asmcli/lib/context.sh

```diff
@@ -0,0 +1,31 @@
+context_init() {
+  local JSON; JSON="${1}"
+  context_FILE_LOCATION="$(mktemp)"; readonly context_FILE_LOCATION
+  export context_FILE_LOCATION
+
+  echo "${JSON}" >> "${context_FILE_LOCATION}"
+}
+
+context_get-option() {
+  local OPTION; OPTION="${1}"
+
+  jq -r --arg OPTION "${OPTION}" '.flags[$OPTION]' "${context_FILE_LOCATION}"
+}
+
+context_set-option() {
+  local OPTION; OPTION="${1}"
+  local VALUE; VALUE="${2}"
+  local TEMP_FILE; TEMP_FILE="$(mktemp)"
+
+  jq --arg OPTION "${OPTION}" --arg VALUE "${VALUE}" \
```

Comment:
Can we try and use the `-S` flag when writing values from `jq` so that the keys are sorted? If it doesn't work then let's not block this PR on it, but I think it will be useful to have deterministic output to help debugging and testing.

Reply:
Will try.

Reply:
It worked. Updated the `-S` flag.